v1/src/lib.rs

```diff
@@ -418,24 +412,29 @@ impl Storage for InfluxDbStorage {
         }
 
         // encode the value as a string to be stored in InfluxDB, converting to base64 if the buffer is not a UTF-8 string
-        let (base64, strvalue) = match String::from_utf8(value.payload.contiguous().into_owned()) {
-            Ok(s) => (false, s),
-            Err(err) => (true, b64_std_engine.encode(err.into_bytes())),
-        };
+        let (base64, strvalue) =
+            match String::from_utf8(ZBuf::from(value.payload()).contiguous().into_owned()) {
+                Ok(s) => (false, s),
+                Err(err) => (true, b64_std_engine.encode(err.into_bytes())),
+            };
 
         // Note: tags are stored as strings in InfluxDB, while fileds are typed.
         // For simpler/faster deserialization, we store encoding, timestamp and base64 as fields.
         // while the kind is stored as a tag to be indexed by InfluxDB and have faster queries on it.
+        let encoding_string_rep = value.encoding().clone().to_string(); // TODO: This i am not entirely sure about
+        let encoding: Encoding = (value.encoding().clone()).into();
+
         let query = InfluxWQuery::new(
             InfluxTimestamp::Nanoseconds(influx_time),
             measurement.clone(),
         )
         .add_tag("kind", "PUT")
         .add_field("timestamp", timestamp.to_string())
-        .add_field("encoding_prefix", u8::from(*value.encoding.prefix()))
-        .add_field("encoding_suffix", value.encoding.suffix())
+        .add_field("encoding_prefix", u16::from(encoding.id()))
```

Comment:
I think we can store just one field "encoding". Maybe we can leave it as is for guarantee compatibility and create issue for replacing this to just "encode" field

Reply:
Yeah, i was worried it would break existing systems to change it. 
I agree that we should rather have a single field> 
We can do this in a future PR.